klippy/extras/display/aip31068_spi.py

```diff
@@ -0,0 +1,206 @@
+# Support for YHCB2004 (20x4 text) LCD displays based on AiP31068 controller
+#
+# Copyright (C) 2024  Alexander Bazarov <oss@bazarov.dev>
+#
+# based on : hd44780_spi.py
+# Copyright (C) 2018  Kevin O'Connor <kevin@koconnor.net>
+# Copyright (C) 2018  Eric Callahan <arksine.code@gmail.com>
+# Copyright (C) 2021  Marc-Andre Denis <marcadenis@msn.com>
+#
+# inspired by https://github.com/red-scorp/LiquidCrystal_AIP31068
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
```

Comment:
Please format the top of the file with 1-line file overview comment, 1-line comment blank, n-line copyrights, 1-line comment blank, 1-line license statement.  It's confusing to list "inspired" by or "based on" comments in the copyright block.  Either some entity has claimed a copyright or not.

Don't feel the need to put my copyright on this file.

Feel free to add contextual information in comments outside of the main copyright block.